crates/viewer/re_viewer/src/app.rs

```diff
@@ -1773,13 +1776,15 @@ impl App {
                 },
             }
 
-            // Do analytics after ingesting the new message,
-            // because that's when the `entity_db.store_info` is set,
-            // which we use in the analytics call.
+            // Do analytics/events after ingesting the new message,
+            // because `entity_db.store_info` needs to be set.
             let entity_db = store_hub.entity_db_mut(store_id);
-            let is_new_store = matches!(&msg, LogMsg::SetStoreInfo(_msg));
-            if is_new_store && entity_db.store_kind() == StoreKind::Recording {
+            if msg_will_add_new_store && entity_db.store_kind() == StoreKind::Recording {
```

Comment:
Before, this was firing off the event twice... Which is really strange. I didn't investigate too deeply, I changed it to check not only that it's a `SetStoreInfo` message, but also that the store actually doesn't exist yet. That fixed it for me. I assume the root cause is that we are sending multiple store info messages somewhere.

Reply:
huh, that's a bit worrisome 🤔
but your check makes sense regardless